src/core/filters/simplefilternode.js

```diff
@@ -73,4 +73,15 @@ export default class SimpleFilterNode extends FilterNode {
   remove () {
     this._remove();
   }
+
+  /**
+   * Returns whether this SimpleFilterNode's filter is equal to another SimpleFilterNode's
+   * @param {SimpleFilterNode} node
+   * @returns {boolean}
+   */
+  hasSameFilterAs (node) {
+    const filter1 = this.getFilter();
+    const filter2 = node.getFilter();
+    return filter1.isEqualToSimpleFilter(filter2);
```

Comment:
Even though isEqualToSimpleFilter is operating on Filter instances, I think it may be better to fold that method into this one here. The Filter class has no concept of "simple". Right now, that concept only makes sense in the context of a FilterNode. If we were writing a generic Filter equality checker, that would definitely live in the Filter class. But, the checker we've added to that class depends on assumptions that are at a different level of abstraction.

Reply:
I was thinking that, eventually we might want to only have "simple" filters and have filter nodes take care of combination. that's not the case right now, though, probably shouldn't be part of this item, and I shouldn't write the SDK into a corner.